pkg/apiclient/http1/facade.go

```diff
@@ -146,3 +157,15 @@ func errFromResponse(r *http.Response) error {
 	}
 	return status.Error(codes.Internal, fmt.Sprintf(": %v", r))
 }
+
+func parseHeaders(headerStrings []string) (http.Header, error) {
+	headers := http.Header{}
+	for _, kv := range headerStrings {
+		items := strings.Split(kv, ":")
```

Comment:
maybe `http` package has header parsing code to use? if not, I'd not bother checking here, just inline the logic at line 103 and keep it simple